text/0000-semicolon-items.md

```diff
@@ -208,6 +258,90 @@ be emitted at once, and not one by one:
 Thereby, the fewer edit-compile cycles will be required and thus the flow
 is disturbed less, but still to a degree.
 
+## Add an `--fpermissive` mode to the compiler
+
+Similar to how `gcc` provides the flag `-fpermissive` which treats *all*
+"sufficiently recoverable" errors as warnings and continues compilation until
+final lib/exe is produced, a Rust compiler could also provide a similar flag.
+This could also be done by writing:
+
+```rust
+#[allow(non_fatal_errors)]
+```
+in the source code.
+
+However, such flags have a few drawbacks:
+* Flags are less likely to be seen as a requirement of the language
+  specification itself. As a consequence, future alternative implementations
+  of Rust will have less of an imperative to support such a flag,
+  making it more likely to develop language dialects, which we've heretofore
+  avoided.
+
+* Not all recoverable warnings are equal, and some may be more indicative of a
+  logic error than a redundant semicolon is. Being able to lint against `;`
+  specifically and configure that is a more limited change.
+
+* The hassle of remembering to use `-fpermissive` or something like it outdoes
+  any gains made by it.
+
+## Do not lint
+
+A more permissive option is to not provide any lint, or make it an
+allow-by-default lint. We reject this because:
+
+1. The following unlikely, but problematic, mistake becomes difficult to spot:
+
+   ```rust
+   #[my_attr]; // <-- hard to spot!
+   fn foo() { ... }
+   ```
+
+   Here, the attribute applies to the semicolon, not the `fn`, but that is
+   not easy to spot at all.
+
+2. We lint today against superfluous punctuation in the form of parenthesis in
+   type and expression contexts. Linting against extraneous semicolons is
+   therefore more consistent.
+
+## Interpret `;` as whitespace at the top level
+
+This solution would simply ignore `;` at the module level or inside `impl`s.
+This will however have too far reaching negative implications on `;` as a
+separator in macros which it is frequently used as.
+
+## Allow an optional `;` after all items
+
+A different technical solution that nonetheless achieves the set out motivations
+of this RFC is to allow syntax like: `struct F {};` specifically. This is then
+parsed as a single item rather than `struct F {}` and `;` as two separate items.
+This is less consistent with the treatment of `;` inside `fn` bodies as well as
+in other languages. 
+
+In addition, the following macro:
+
+```rust
+macro_rules! m {
+    ($($item: item);*) => {}
```

Comment:
I basically agree this is unlikely to occur in the wild and that it will more likely have a matcher `$($item:item)*`.

I am less sure about macros affecting both proposals (as much or in the same way?..).

Reply:
In the same way, more or less the same amount (different specific cases), IMO.